nselib/sasl.lua

```diff
@@ -458,4 +460,37 @@ Helper = {
   end,
 }
 
+if not unittest.testing() then
+  return _ENV;
+end
+
+-- self-testing starts here
+local expected = {}
+
+local object = DigestMD5:new("Digest nonce=\"9e4ab724d272474ab13b64d75300a47b\", opaque=\"de40b82666bd5fe631a64f3b2d5a019e\", realm=\"me@kennethreitz.com\", qop=auth", "tempuname", "tempupasswd", "tempmethod", "/temp/path")
```

Comment:
We generally prefer to have shorter lines. Lua has a nice feature, the `\z` escape, which escapes a sequence of whitespace (newlines, spaces, tabs) within a quoted string. So you could write this as:
```lua
local object = DigestMD5:new('Digest nonce="9e4ab724d272474ab13b64d75300a47b", \z
    opaque="de40b82666bd5fe631a64f3b2d5a019e", \z
    realm="me@kennethreitz.com", qop=auth',
  "tempuname", "tempupasswd", "tempmethod", "/temp/path")
```

I also prefer, as you see, to use single quotes to avoid escaping double quotes. If you need to use both, you can use [lua's `[[long brackets]]` syntax](http://www.lua.org/manual/5.3/manual.html#3.1), too.